pkg/pool-weighted/test/ManagedPool.test.ts

```diff
@@ -388,6 +388,28 @@ describe('ManagedPool', function () {
             bptAmount: actualManagementFeeBpt,
           });
         });
+
+        it('syncs the total supply to the actual supply', async () => {
+          // As we're performing a join or exit here we need to account for the change in the BPT total supply due to
+          // the join/exit. We do this by tracking the user's balance.
+          const balanceBefore = await pool.balanceOf(other);
+          const totalSupplyBefore = await pool.totalSupply();
```

Comment:
I think these tests are doing too many things. I'd have a) one test that checks getActualSupply returns the correct value, b) one that checks that it gets synced after the operation, and c) one that checks that the operation mints the difference between virtual and actual

Reply:
(test c) is more of a maybe)